File: android-app/app/src/main/java/com/sherlock/android/model/CheckResult.kt
package com.sherlock.android.model

enum class ResultStatus { FOUND, NOT_FOUND, ERROR }

data class CheckResult(
    val siteName: String,
    val url: String,
    val status: ResultStatus,
    val errorMessage: String? = null,
    val isDanishDating: Boolean = false
)

File: mobile-app/lib/models/check_result.dart
enum ResultStatus { found, notFound, error }

class CheckResult {
  final String siteName;
  final String url;
  final ResultStatus status;
  final String? errorMessage;

  const CheckResult({
    required this.siteName,
    required this.url,
    required this.status,
    this.errorMessage,
  });
}
